### ISSUE TITLE
Fails when targeting ES2015 in TypeScript

### ISSUE THREAD

**Issue by reduckted** [mgechev/injection-js#48]
This library was working beautifully in the TypeScript application I was writing (thanks! 😁) that targeted ES5. I wanted to use private identifiers which requires targeting ES2015 or higher. So I switched the target to ES2015 and everything broke.

Specifically, this line fails because `prototype` is undefined when using a factory provider (`useFactory`):
https://github.com/mgechev/injection-js/blob/dca2b5dbb0ece3a11cf0b777b8154df15f1c5d19/lib/reflection/reflection_capabilities.ts#L262

I've tried using `@abraham/reflection`, `reflect-metadata` and `core-js/es7/reflect` and all had the same outcome. This is my TypeScript config:
```json
{
    "compilerOptions": {
        "lib": ["ES2015"],
        "target": "ES2015",
        "module": "CommonJS",
        "strict": true,
        "esModuleInterop": true,
        "experimentalDecorators": true,
        "emitDecoratorMetadata": true
    }
}
```

I can put together a small reproduction if needed. I just thought I'd check here first in case this is a known limitation.

**Comment by reduckted** [comment]
Here's a simple reproduction: https://github.com/reduckted/repro-injection-js-48

**Comment by mgechev** [comment]
Just published 2.3.2 which fixes the issue.

**Comment by reduckted** [comment]
Works great. Thank you so much! ❤️
